src/bundles/peer-locations.js

```diff
@@ -300,44 +141,67 @@ class PeerLocationResolver {
       ...opts.cache
     })
 
+    this.failedAddrs = HLRU(500)
+
+    this.queue = new PQueue({
+      concurrency: opts.concurrency,
+      autoStart: true
+    })
+
     this.geoipLookupPromises = {}
   }
 
-  async lookupWithCaches (peerId, addr, getIpfs) {
-    const ipv4Tuple = Multiaddr(addr).stringTuples().find(isNonHomeIPv4)
-    if (!ipv4Tuple) {
-      throw new Error(`Unable to resolve location for non-IPv4 address ${addr}`)
-    }
+  async findLocations (peers, getIpfs) {
+    const res = {}
 
-    var ipv4Addr = ipv4Tuple[1]
+    for (const p of peers) {
+      const peerId = p.peer.toB58String()
+      const addr = p.addr.toString()
 
-    // maybe we have it cached by ipv4 address already, check that.
-    let location = await this.geoipCache.get(ipv4Addr)
-    if (location) {
-      return location
-    }
+      const ipv4Tuple = Multiaddr(addr).stringTuples().find(isNonHomeIPv4)
```

Comment:
I think we can just use multiaddr instance directly?
(Or was this a workaround for some bug?)

```suggestion
      const ipv4Tuple = p.addr.stringTuples().find(isNonHomeIPv4)
```